src/commands/synthetics/__tests__/utils.test.ts

```diff
@@ -969,10 +1114,13 @@ describe('utils', () => {
     })
 
     test('correct number of pass and timeout results', async () => {
-      const pollTimeoutResult: PollResult = {...pollResult, resultID: 'another-id'}
+      const pollTimeoutResult: PollResult = {...deepExtend({}, pollResult), resultID: 'another-id'}
```

Comment:
What's the intention of the changes to this test? It seems unrelated to the in-progress logging.

Reply:
I fixed this test as a separate commit first, before doing any code change: https://github.com/DataDog/datadog-ci/pull/1097/commits/f9aa266004b380e9da73317ec6a56ff58b1dc5ba

This test was buggy because this line of code makes side effects:
https://github.com/DataDog/datadog-ci/blob/d045ad3aada023ff6c7d334cb2fc1fe2a1427583/src/commands/synthetics/utils.ts#L331-L332

This side effect was setting the `failure` property in **both the passed `pollResult`** and failed `pollTimeoutResult` here: https://github.com/DataDog/datadog-ci/blob/f9aa266004b380e9da73317ec6a56ff58b1dc5ba/src/commands/synthetics/__tests__/utils.test.ts#L983

The rest of the edits made to this test is to make it more readable.

Reply:
I had to fix this test because it started failing **_because of the side effect which makes it buggy_** once I made my code changes.

Reply:
👍 sounds good, thanks for the explanation